examples_tests/22.RaytracedAO/Renderer.cpp

```diff
@@ -198,23 +214,55 @@ Renderer::InitializationData Renderer::initSceneObjects(const SAssetBundle& mesh
 					return std::hash<decltype(view)>()(view);
 				}
 			};
-			core::unordered_map<VisibilityBufferPipelineKey, core::smart_refctd_ptr<ICPURenderpassIndependentPipeline>, VisibilityBufferPipelineKeyHash> visibilityBufferFillPipelines;
+			core::unordered_map<VisibilityBufferPipelineKey,core::smart_refctd_ptr<ICPURenderpassIndependentPipeline>,VisibilityBufferPipelineKeyHash> visibilityBufferFillPipelines;
 
 			{
 				meshesToProcess.reserve(contents.size());
+				meshBuffersToProcess.reserve(contents.size());
 
-				uint32_t drawableCount = 0u;
+				uint32_t drawableCount = 0u; // TODO: redo
 				for (const auto& asset : contents)
 				{
 					auto cpumesh = static_cast<asset::ICPUMesh*>(asset.get());
 					auto meshBuffers = cpumesh->getMeshBuffers();
 					assert(!meshBuffers.empty());
 
+					auto& meshbuffers = cpumesh->getMeshBufferVector();
+					for (auto mbIt = meshbuffers.begin(); mbIt!=meshbuffers.end(); mbIt++)
+						meshBuffersToProcess.push_back(mbIt->get());
+
 					meshesToProcess.push_back(cpumesh);
 
-					drawableCount += meshBuffers.size()*retval.globalMeta->getAssetSpecificMetadata(cpumesh)->m_instances.size();
+					drawableCount += meshBuffers.size()*retval.globalMeta->getAssetSpecificMetadata(cpumesh)->m_instances.size(); // TODO: redo
 				}
-				cullData.resize(drawableCount);
+				cullData.resize(drawableCount); // TODO: redo
+			}
+			// split into packed batches
+			{
+				constexpr uint16_t minTrisBatch = 256u; 
+				constexpr uint16_t maxTrisBatch = MAX_TRIANGLES_IN_BATCH;
+				constexpr uint8_t minVertexSize = 
+					asset::getTexelOrBlockBytesize<asset::EF_R32G32B32_SFLOAT>()+
+					asset::getTexelOrBlockBytesize<asset::EF_A2R10G10B10_SNORM_PACK32>()+
+					asset::getTexelOrBlockBytesize<asset::EF_R32G32_SFLOAT>();
+
+				constexpr uint8_t kIndicesPerTriangle = 3u;
+				constexpr uint16_t minIndicesBatch = minTrisBatch*kIndicesPerTriangle;
+				
+				CPUMeshPacker::AllocationParams allocParams;
+				allocParams.vertexBuffSupportedByteSize = 1u<<31u;
+				allocParams.vertexBufferMinAllocByteSize = minTrisBatch*minVertexSize;
+				allocParams.indexBuffSupportedCnt = (allocParams.vertexBuffSupportedByteSize/allocParams.vertexBufferMinAllocByteSize)*minIndicesBatch;
+				allocParams.indexBufferMinAllocCnt = minIndicesBatch;
+				allocParams.MDIDataBuffSupportedCnt = allocParams.indexBuffSupportedCnt/minIndicesBatch;
+				allocParams.MDIDataBuffMinAllocCnt = 1u; //so structs from different meshbuffers are adjacent in memory
+    
+				auto cpump = core::make_smart_refctd_ptr<CCPUMeshPackerV2<>>(allocParams,minTrisBatch,maxTrisBatch);
+
+				core::vector<CPUMeshPacker::ReservedAllocationMeshBuffers> allocData(cpump->calcMDIStructMaxCount(meshBuffersToProcess.begin(),meshBuffersToProcess.end()));
+				printf("%d\n",allocData.size());
+				cpump->alloc(allocData.data(),meshBuffersToProcess.begin(),meshBuffersToProcess.end());
+				cpump->shrinkOutputBuffersSize();
```

Comment:
@Przemog1 I need the TODOs about shrinking (and other allocator stuff) to be addressed
(merge my pathtracing branch into yours and fix this up quickly)